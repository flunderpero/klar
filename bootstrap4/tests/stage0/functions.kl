--- expected

# compiler output:
# exit code:
0
# stderr:
# stdout:
1
2
3
4
6

/expected ---

fn foo():
    print(1)
end

fn foo_with_arg(x i32):
    print(x)
end

fn foo_with_multiple_args(x i32, y i32):
    print(x)
    print(y)
end

fn foo_with_return_value(x i32) i32:
    return x + 1
end

fn main():
    foo()
    foo_with_arg(2)
    foo_with_multiple_args(3, 4)
    print(foo_with_return_value(5))
end
