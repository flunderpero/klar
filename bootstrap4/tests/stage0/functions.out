1
2
3
4
---
exit code: 0
